### PR TITLE
Addresses review comments in #309

### PR DESCRIPTION
**Pull request by pksunkara**
Also includes minor cleaning of trailing whitespaces and removal of unnecessary tests.

### CODE REVIEW

File: test/test-ActionParser.cc [hunk 1]
```diff
@@ -490,7 +490,7 @@ TEST_CASE("Warn when request is not followed by a response", "[action]")
     REQUIRE(action.sourceMap.examples.collection[1].responses.collection.size() == 0);
 }
 
-// TODO: to be implemented
+// TODO: This test is failing because of a bug in action attributes. Need to fix the bug.
```
Comment (zdne) [inline test/test-ActionParser.cc]: Any link to the bug? Is it noted in Trello, wiki or issues?

File: src/MSONParameterParser.h [hunk 2]
```diff
@@ -178,6 +181,8 @@ namespace snowcrash {
                     }
                     else {
                         if (RegexMatch(attributes[i], "^enum\\[[^][]+]$")) {
+
+                            // Retrieve the type which is wrapped by enum[]
                             out.node.type = attributes[i].substr(5, attributes[i].length() - 6);
```
Comment (zdne) [inline src/MSONParameterParser.h]: I would still love to explain `5` and `6` :) looks magical

File: test/test-ActionParser.cc (hunk 1 above)
Comment (pksunkara) [inline test/test-ActionParser.cc]: It is in Trello product inbox.

File: src/MSONParameterParser.h (hunk 2 above)
Comment (pksunkara) [inline src/MSONParameterParser.h]: The length of `enum[` is 5 and `enum[]` is 6. :smiling_imp:

Comment (zdne) [inline src/MSONParameterParser.h]: Hmm will that work with `enum    [string   ]` ?

Comment (pksunkara) [inline src/MSONParameterParser.h]: I can `TrimString(out.node.type)`.

**Comment by pksunkara** [comment]
Updated.

File: src/MSONParameterParser.h (hunk 2 above)
Comment (zdne) [inline src/MSONParameterParser.h]: I think it wold be good if it is tolerant to white spaces. Also please replace the numbers with some constants + comments. We really had no idea whats up reading these numbers.

Comment (pksunkara) [inline src/MSONParameterParser.h]: Already made it tolerant to white spaces. Regarding constants, I don't agree with that. We are stripping `enum[` and `]` which are hardcoded and therefore I hardcoded the numbers `5` and `6` just like how we hardcoded some numbers in MSON parsing for parsing some punctuations.

Comment (zdne) [inline src/MSONParameterParser.h]: > Regarding constants, I don't agree with that

There is nothing to agree or disagree here. These are magical numbers without any obvious relation to the problem. Please refactor them into constants. See "Accepted limited use of magic numbers" in http://en.wikipedia.org/wiki/Magic_number_(programming) and / or http://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad

Using an arbitrary numbers like this in the hope that it is obvious that some other constant somewhere is related to this is really against readability, clarity and maintainability.

**Comment by pksunkara** [comment]
Updated.

File: src/MSONParameterParser.h [hunk 3]
```diff
@@ -177,8 +183,13 @@ namespace snowcrash {
                         definedUse = true;
                     }
                     else {
-                        if (RegexMatch(attributes[i], "^enum\\[[^][]+]$")) {
-                            out.node.type = attributes[i].substr(5, attributes[i].length() - 6);
+
+                        // Retrieve the type which is wrapped by enum[]
+                        std::string typeInsideEnum = RegexCaptureFirst(attributes[i], EnumRegex);
```
Comment (zdne) [inline src/MSONParameterParser.h]: :+1:

**Pull request merged**